### PR TITLE
Reduce packaged size of extensions

### PR DESCRIPTION
**Pull request by anthonydresser**
Some additional work could be done to azurecore, but this was the low hanging fruit

Notebooks
Before:
image

After:
image

Azure Core
Before:
image

After: 
image

Mssql
Before:
image

After:
image

### CODE REVIEW

**Comment by kevcunnane** [comment]
Nice - I was going to start on this after the meeting yesterday but you beat me to it 😄 

Can you share a build with this around for testing before we integrate it? Things I want to verify:
- Jupyter config files are still found & copied on Notebook startup. @corivera, @rajmusuku / @chlafreniere please help with verification
- Config files are found, specifically the one used by Jupyter Python installer.
- Similarly for mssql no issues with config or other files are found.

File: extensions/azurecore/extension.webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+const withDefaults = require('../shared.webpack.config');
+
+module.exports = withDefaults({
+	context: __dirname,
+	entry: {
+		extension: './src/extension.ts'
+	},
+	externals: {
+		'ms-rest': 'commonjs ms-rest',
+		'request': 'commonjs request'
```
Comment (kevcunnane) [inline extensions/azurecore/extension.webpack.config.js]: Out of interest how did you decide which dependencies to put into their own 'require' files? Could we save on loading a lot of lodash stuff by referencing that as an external too?

File: extensions/azurecore/package.json [hunk 2]
```diff
@@ -164,17 +164,18 @@
 		"hasAzureResourceProviders": true
 	},
 	"dependencies": {
+		"adal-node": "^0.1.28",
```
Comment (kevcunnane) [inline extensions/azurecore/package.json]: This is to fix packaging issues where webpack didn't detect the transitive dependency?

File: extensions/mssql/extension.webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+module.exports = withDefaults({
+	context: __dirname,
+	entry: {
+		extension: './src/main.ts'
+	},
+	externals: {
+		'clipboardy': 'commonjs clipboardy',
+		'dataprotocol-client': 'commonjs dataprotocol-client',
+		'vscode-languageclient': 'commonjs vscode-languageclient'
```
Comment (kevcunnane) [inline extensions/mssql/extension.webpack.config.js]: FYI we also use the request module here. Should we make it an external like in azurecore?

Comment (kevcunnane) [reply extensions/mssql/extension.webpack.config.js]: Just checked - Gene put it in the wrong package.json. I'll have a PR out that fixes insiders build ASAP, please update this once it's in.

File: extensions/azurecore/extension.webpack.config.js (hunk 1 above)
Comment (anthonydresser) [reply extensions/azurecore/extension.webpack.config.js]: As far as I found lodash was only used by mssql for the decompress stuff for the service-downloader. How vscode's build scripts work you can only specify external dependency that you directly reference in the package.json. I put external dependencies as any that it, or one of it's dependencies, doesn't behave correctly for webpack to work.

File: extensions/azurecore/package.json (hunk 2 above)
Comment (anthonydresser) [reply extensions/azurecore/package.json]: Either it, or one of it's dependencies using expression based require which webpack can't read.

**Comment by kburtram** [comment]
@anthonydresser is this PR ready?  If we want this to target March we should merge soon to include in test passes.

**Comment by anthonydresser** [comment]
It still needs testing, unless we really want it for march I would say lets skip this for march.

**Comment by kburtram** [comment]
@anthonydresser sounds good.  April is fine, I'm just going through pending PRs to make sure changes targeting March get merged in a timely manner.

**Comment by anthonydresser** [comment]
@kevcunnane @chlafreniere Can someone from the notebook guys help me test this branch and make sure your stuff is still working in a packaged build?

**Pull request merged**